accepted/2021/winforms/streamline-application-bootstrap.md

```diff
@@ -0,0 +1,302 @@
+Microsoft, these roles happen to match to job titles too, but that's irrelevant.
+-->
+
+**Owner** [Igor Velikorossov](https://github.com/RussKie)
+
+# Overview
+
+Historically Windows Forms designer was rendering its design surface based on hardcoded assumptions, such as control default font and dpi settings. However over the past 10-15 years not only multi-monitor configurations became a norm, monitor tech has made significant progress in making 4K, 5K, 8K, etc. monitors widely used by our consumers. The Windows itself has been innovating and tinkering with UI configurations, such default font metrics, which over the years deviated from "Microsoft Sans Serif, 8pt" and now is set to "Segoe UI, 9pt". Incidentally the office team is currently [in search of the next default font](https://www.microsoft.com/microsoft-365/blog/2021/04/28/beyond-calibri-finding-microsofts-next-default-font/).
+
+The .NET Core/.NET-powered Windows Forms runtime has been (more or less) keeping up with the changes, e.g. by providing new API to [set high dpi mode](https://github.com/dotnet/winforms/pull/396), or by updating the [default font](https://github.com/dotnet/winforms/pull/656). The default font change has unearthed numerous issues in the runtime (e.g. different font-based scaling), which we addressed by providing (yet another) API to set an [application-wide default font](https://github.com/dotnet/winforms/pull/4911). 
+
+However during numerous discussions the team has identified probably the biggest flaw in the current separation of the runtime and the designer, and the overall move from .NET Framework assumptions - the lack of a mechanism to share Windows Forms project-level configuration settings between the runtime and the designer. That is, if the app is configured to run with disabled visual styles, in PerMonitorV2, and have an application-wide default font set to "Arial, 14pt" there is no way to show the designer surface with the same settings to truly provide the WYSIWYG experience.
+
+
+
+# Proposal
+
+The purpose of this proposal is to:
+
+1. **streamline a Windows Forms application configuration and bootstrap**, 
+2. with the view that this will facilitate the **sharing of the information between the runtime and the designer during the development phase**.<br/>
+That is, whenever the designers surface process is started configuration information is read from a known location, and necessary configurations are applied (e.g. run the design surface in PerMonitorV2 mode, or set a form/usercontrol default font to "Arial, 14pt").<br/>
+:warning:  **out of scope**, tracked under https://github.com/dotnet/winforms-designer/issues/3192.
+
+**NOTE:** The new functionality is opt-in, i.e. unless a developer makes a conscious decision to use the new configuration and the bootstrap mechanism existing applications will continue to work as-is, and the current developer experience will remain the same.
```

Comment:
Do you actually mean opt-in, or do you mean "non-breaking"?  I presume we'd want to update templates to start with this for example?

Reply:
Yes, the plan is to update templates. For existing apps the behaviour is "opt-in" in the sense that developers will need to write the new method, and remove existing `Application.*` methods. If they don't do this - nothing happens, apps will continue to work as before.